gril/grilrequest.c

```diff
@@ -807,24 +807,45 @@ void g_ril_request_set_smsc_address(GRil *gril,
 }
 
 void g_ril_request_dial(GRil *gril,
-			const struct ofono_phone_number *ph,
-			enum ofono_clir_option clir,
-			struct parcel *rilp)
+                      const struct ofono_phone_number *ph,
+                      gboolean isEmergency,
+                      enum ofono_clir_option clir,
+                      struct parcel *rilp)
 {
-	parcel_init(rilp);
-
-	/* Number to dial */
-	parcel_w_string(rilp, phone_number_to_string(ph));
-	/* CLIR mode */
-	parcel_w_int32(rilp, clir);
-	/* USS, empty string */
-	/* TODO: Deal with USS properly */
-	parcel_w_int32(rilp, 0);
-	parcel_w_int32(rilp, 0);
+      const char *phoneNoString;
+      parcel_init(rilp);
+      phoneNoString = phone_number_to_string(ph);
+
+      /* Number to dial */
+      parcel_w_string(rilp, phoneNoString);
+      /* CLIR mode */
+      parcel_w_int32(rilp, clir);
+      if (g_ril_vendor(gril) == OFONO_RIL_VENDOR_SAMSUNG_MSM_822x) {
+          parcel_w_int32(rilp, 0); // CallDetails.call_type
+          if (isEmergency) {
+              parcel_w_int32(rilp, 3); // Emergency call domain?
+          } else {
+              parcel_w_int32(rilp, 1); // Default call domain?
+          }
+          parcel_w_string(rilp, ""); // CallDetails.getCsvFromExtras
+      }
+      /* USS, empty string */
+      /* TODO: Deal with USS properly */
+      parcel_w_int32(rilp, 0);
+
+      g_ril_append_print_buf(gril, "(%s,%d,0,0)",
+                              phoneNoString,
+                              clir);
+}
```

Comment:
Seems like you inadvertently change the whitespace here. Remember, oFono uses tab characters, not spaces.